File: DataStructures/src/main/java/com/zhangdd/sort/BubbleSort.java
package com.zhangdd.sort;

/**
 * @author zhangdd on 2021/10/19
 */
public class BubbleSort {

    public static void main(String[] args) {
        //从小到大排
        int[] arr = {1, 9, 7, -3, 2};

        for (int i = 0; i < arr.length - 1; i++) {
            int temp;
            for (int j = 0; j < arr.length - 1 - i; j++) {
                if (arr[j] > arr[j + 1]) {
                    temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }

        for (int j : arr) {
            System.out.printf("%d ", j);
        }
    }
}

File: python/Algorithm/sort/BubbleSort.py
#!/usr/bin/env python3
"""
@author: zhangdd on 2022/2/3
"""


def bubble_sort(arr):
    for i in range(len(arr) - 1):
        for j in range(len(arr) - 1 - i):
            if arr[j] > arr[j + 1]:
                tmp = arr[j]
                arr[j] = arr[j + 1]
                arr[j + 1] = tmp


if __name__ == '__main__':
    collection = [1, 9, 7, -3, 2]
    bubble_sort(collection)
    print(collection)
